src/zenml/cli/utils.py

```diff
@@ -115,15 +115,37 @@ def pretty_print(obj: Any) -> None:
     click.echo(str(obj))
 
 
-def echo_component_list(component_list: Mapping[str, "BaseComponent"]) -> None:
-    """Echoes a list of components in a pretty style."""
+def print_table(obj: List[Dict[str, str]]) -> None:
+    """Echoes the list of dicts in a table format. The input object should be a
+    List of Dicts. Each item in that list represent a line in the Table. Each
+    dict should have the same keys. The keys of the dict will be used as
+    headers of the resulting table.
+
+    Args:
+      obj: An List containing dictionaries.
+    """
+    click.echo(tabulate(obj, headers="keys"))
+
+
+def format_component_list(
+    component_list: Mapping[str, "BaseComponent"]
+) -> List[Dict[str, str]]:
+    """Formats a list of components into a List of Dicts. This list of dicts
+    can then be printed in a table style using cli_utils.print_table.
+
+    Args:
+      component_list: The component_list is a mapping of component key to
+                      component class with its relevant attributes
+    Returns:
+        list_of_dicts: A list of all components with each component as a dict
+    """
     list_of_dicts = []
     for key, c in component_list.items():
         # TODO [ENG-143]: This will break if component has element called `key`
```

Comment:
Can we also deal with this problem somehow? Maybe by using a word which isnt so common like "key"?

Reply:
I will rename `key` to `name`.

Reply:
or is that also too common? Maybe component_name?

Reply:
`item`?

Reply:
`name` does feel most natural

Reply:
How about _name? Or name_?

Reply:
GitHub formatted it. I meant adding a leading or trailing underscore

Reply:
hmm, what do you think of this. I used name, and in case the component has an attribute name is rename that attribute to component name for the sake of the printout